flink-core/src/main/java/org/apache/flink/api/common/state/StateTtlConfig.java

```diff
@@ -327,6 +329,14 @@ public Builder cleanupInRocksdbCompactFilter(long queryTimeAfterNumEntries) {
 			return this;
 		}
 
+		/**
+		 * Cleanup expired state by activating both strategies. Depending on actually used backend, the corresponding cleanup will kick in.
+		 */
+		@Nonnull
+		public Builder cleanupInBackground() {
+			return cleanupIncrementally(10, true).cleanupInRocksdbCompactFilter(1000L);
```

Comment:
I don't think the setting should delegate to the specific options. This will mean that this code will have to be adapted in the future if other backends or other cleanup strategies are added.

The state backends should notice that "cleanup in background" is enabled and then react accordingly.

Reply:
@aljoscha so you mean we do not need this `cleanupInBackground` and stay the same?

Reply:
@aljoscha true, we can move concrete decision about default configuration to backends.

@yanghua the method `cleanupInBackground` should be  still added but rather as a boolean flag in StateTtlConfig. Backends should check this flag where they check configuration of their corresponding cleanup strategies and use default ones.

Reply:
exactly! 👌

Reply:
Hi @aljoscha and @azagrebin I have refactored the code and give a **non-final** version of PR for discussing. If we let specific statebackend react for the special strategy, we may change some class's access modifier. WDYT?